Polyconseil/easygettext
# Support extraction of $gettext from vue <script> tags

We would like to have the possibility to extract localized strings from the script tags in Vue.js files when we use vue-gettext.
Using xgettext is painful and subject to many restrictions:
- It is not maintained since a long time (at least for js)
- It does not parse modern ECMAScript features like template strings.
- Parsing .vue files is possible but requires a few hacks like adding parenthesis around `<template>` and `<script>` and breaks mysteriously anyway.

The goal of this commit is to introduce the possibility to extract localized strings located in Vue.js `<script>` tags. This is a first step. For the moment, only strings marked as translatable using the $gettext function are extracted.

Extraction support for other functions like $ngettext, $pgettext or $npgettext could come later.
If you have any question or feedback, please let me know!

How to test:
- Setup a .vue file like
```
    <template>
        <h1>{{ greeting_message }}</h1>
    </template>
    <script>
        export default {
            name: "greetings",
            computed: {
                greeting_message() {
                    return this.$gettext("Hello there!")
                },
                duplicated_greeting_message() {
                    return  this.$gettext("Hello there!")
                }
            }
        }
     </script>
```

- Run compile-cli.js with this file

Expected result -> Strings present in the script part of the vue
component have been successfully extracted.
